lcd.lua

```diff
@@ -171,6 +171,44 @@ local generate_texture = function(lines)
 	return texture
 end
 
+local number_to_color = function(num)
+	if type(num) ~= "number" or num >= 4096 or num < 0 then
+		return "#0000"
```

Comment:
In case of an error, it might make more sense to either:
 * fall back to `0xF00` (red = error)
 * or convert to 0x000 (black = error)
 * or pass `tonumber(num) or 0` to the numeric computation below and let the user see when it looks strange.

Also: What if `nan` is passed to this function?